pkg/metrics/metrics_test.go

```diff
@@ -23,7 +30,15 @@ func TestStart(t *testing.T) {
 	}
 	metricsServer := NewServer(opts)
 
-	go metricsServer.Start()
+	g.Go(func() error {
+		err := metricsServer.Start(ctx)
+		if err != nil {
+			t.Errorf("Expected err to be nil: %q", err)
+			return err
+		}
+
+		return nil
+	})
```

Comment:
Yes, this is something that I've been thinking about. Wasn't able to figure out where this would fit - but maybe is enough to create a helper package and place the start/stop functions and interfaces there? Not sure, but I would like to reuse them inside of the tests.